lib/index.js

```diff
@@ -507,14 +527,13 @@ Socket.prototype._flush = function() {
     };
 
     if (flags & zmq.ZMQ_POLLIN) {
-      emitArgs = ['message'];
+      emitArgs = this._read();
 
```

Comment:
I've just extracted the reading part from _flush into _read. We could make it "public" and rename it to **read**